### ISSUE TITLE
<referencegroup> vs RFC style requirements

### ISSUE THREAD

**Issue by reschke** [rfc-format/draft-iab-xml2rfc-v3-bis#10]
(from https://tools.ietf.org/tools/xml2rfc/trac/ticket/323)

I don't believe <referencegroup> as currently defined in the vocab and HTML specs addresses the RFC Style requirement for citing BCPs/STDs.

Either we need to fix <referencegroup> or change/clarify the RFC Style document.

**Comment by paulehoffman** [comment]
I agree that this needs fixing in the v3 format, although it might require fixes to both documents.

**Comment by reschke** [comment]
Further issues:

- we need to define whether the members of the referencegroup can be referenced; and how that would look like
- we need to define how referencing the referencegroup looks like (for all <reference> formats other than the default)
- we need to define whether sorting references affects the contents if reference groups

**Comment by stpeter** [comment]
@reschke https://trac.ietf.org/trac/xml2rfc/ticket/323 says "overtaken by events" ... is this still a live issue?

**Comment by reschke** [comment]
Am 17.06.2021 um 17:17 schrieb Peter Saint-Andre:
> @reschke <https://github.com/reschke>
> https://trac.ietf.org/trac/xml2rfc/ticket/323
> <https://trac.ietf.org/trac/xml2rfc/ticket/323> says "overtaken by
> events" ... is this still a live issue?

Likely, but I'll have to research. Give me a few days.

**Comment by reschke** [comment]
First of all, https://github.com/rfc-format/draft-iab-xml2rfc-v3-bis/issues/10#issuecomment-461593132 still seems to be relevant.

Furthermore, I did a quick comparison of https://datatracker.ietf.org/doc/html/draft-flanagan-7322bis-07#section-4.8.6.2 with a recent example (https://www.rfc-editor.org/rfc/rfc8722.html#BCP9), and xml2rfc indeed seems to do what the style guide says.

That said, in addition to the comments above:

- I don't get why we don't allow a reference group to have a title (this may not be relevant for IETF BCP/STD, but could be for other references)
- Having the rendering of the entries differ so much from "regular" rendering (no anchors, no hyperlinks, DOI suppressed) doesn't make any sense to me.

At a minimum, we need to define how rendering of \<reference\> entries inside a group differs from normal rendering, and that needs to be defined for non-IETF reference groups as well.

**Comment by reschke** [comment]
FWIW, I also believe that the rendering should pull out the series number and make it an item of the group.

**Comment by cabo** [comment]
Archived-At: <https://mailarchive.ietf.org/arch/msg/xml2rfc/Dj9zM4ZGuonS236NQ0sR7Rr0ai0>

I think the main conclusion is that the rendering for a reference to an anchor *inside* a <referencegroup needs to reveal the referencegroup, as in

```
     See RFC 1034 [STD13].
```

(see RFC 7322).

**Comment by cabo** [comment]
This is compounded by the fact that xref rendering requires the author to be aware whether a parenthetical form will be generated (as in `[RFC1234]`) or not (as in `Section 3`), which requires different text around the xref rendering.
As an author, I run into this all the time, and it is not pretty.  But maybe we should fix the above problem first.

**Comment by reschke** [comment]
I believe that for each reference in a reference group, the anchor needs to be inserted in the visible text (nested). Once we have that, it becomes clear how citing something inside a reference group will work.

(Yes, that requires changes to multiple specs)

**Comment by ajeanmahoney** [comment]
@reschke Would nested anchors in a subseries reference look something like the following?

```
   [BCP13]    "<title>", Month YYYY

              [RFC4289] Freed, N. and J. Klensin, "Multipurpose Internet Mail
                        Extensions (MIME) Part Four: Registration Procedures",
                        BCP 13, RFC 4289, DOI 10.17487/RFC4289, December 2005.
                        <https://www.rfc-editor.org/info/rfc4289>.

              [RFC6838] Freed, N., Klensin, J., and T. Hansen, "Media Type
                        Specifications and Registration Procedures", BCP 13,
                        RFC 6838, DOI 10.17487/RFC6838, January 2013.
                        <https://www.rfc-editor.org/info/rfc6838>.

              <https://www.rfc-editor.org/info/bcp13>
```

The above is just my attempt (with my individual hat on) to sketch what a reference to a subseries could look like if a subseries had a title and if the RFC information also included DOIs and URLs. I'm not sure if Month YYYY should show when the subseries was created, or last updated, or if it should even exist. 

For comparison, this is how the entry for [BCP13] currently looks:

```
   [BCP13]    Freed, N. and J. Klensin, "Multipurpose Internet Mail
              Extensions (MIME) Part Four: Registration Procedures",
              BCP 13, RFC 4289, December 2005.

              Freed, N., Klensin, J., and T. Hansen, "Media Type
              Specifications and Registration Procedures", BCP 13,
              RFC 6838, January 2013.

              <https://www.rfc-editor.org/info/bcp13>
```

Note that RFC 7322 says:
```
   *  References and citations must match.  That is, there must be a
      reference for each citation used, and vice versa.
```
And xml2rfc enforces this. This may be a reason why the RFC entries within a subseries reference do not display anchors (I've been digging through archives, but I haven't found an explanation of why the subseries reference is constructed the way it is).

**Comment by cabo** [comment]
On 13. Feb 2023, at 22:14, Jean Mahoney ***@***.***> wrote:
> 
> I haven't found an explanation of why the subseries reference is constructed the way it is).

To me it always seemed like a PoC that got deployed a bit too early.

Yes, the constituent documents need an anchor, and that needs to be visible, so I generally like your design.
But what do you put under "<title>", Month YYYY?
Do BCPs have titles and dates?

Grüße, Carsten

**Comment by ajeanmahoney** [comment]
> But what do you put under "<title>", Month YYYY?

Good question, @cabo. Currently BCPs and STDs don't have either titles or dates. I'm not sure a date is meaningful, but it might be nice for BCPs and STDs to have titles. 

There wasn an interesting issue that was raised over in bibxml-service (https://github.com/ietf-tools/bibxml-service/issues/303) about BCP 113 no longer existing. Maybe if BCPs and STDs had titles, we could say something like BCP 113 is empty rather than [it doesn't exist](https://www.rfc-editor.org/info/bcp113).

**Comment by reschke** [comment]
@ajeanmahoney - thanks a lot.

Yes, that's the general layout I had in mind and which I would have used.

Regarding your questions:

1. I believe some BCPs have titles. If that's the case, we should display them (and the xml2rfc machinery to use them in \<xref>s should work).
2. I don't believe they have dates, but if they did, that's where they should go. That said, the \<referencegroup> element could be used for other document series as well, and there's no reason it couldn't have a date in that case. (FWIW, it also could have authors etc..)
3. Having too many anchors is not an hard error bust just a warning (IMHO), but not displaying anchors that can actually be used /is/ a problem. We can relax the warning for unused references if they appear inside a \<referencegroup>.

The reason the format is like it this is IMHO because that's what Joe Hildebrand suggested in them HTML format spec, and it never was discussed properly (ahem).

There's one thing we could try for minimizing the output and pulling out common seriesInfo elements, like that:

~~~
   [BCP13]    "<title>", Month YYYY, BCP 13

              [RFC4289] Freed, N. and J. Klensin, "Multipurpose Internet Mail
                        Extensions (MIME) Part Four: Registration Procedures",
                        RFC 4289, DOI 10.17487/RFC4289, December 2005.
                        <https://www.rfc-editor.org/info/rfc4289>.

              [RFC6838] Freed, N., Klensin, J., and T. Hansen, "Media Type
                        Specifications and Registration Procedures",
                        RFC 6838, DOI 10.17487/RFC6838, January 2013.
                        <https://www.rfc-editor.org/info/rfc6838>.

              <https://www.rfc-editor.org/info/bcp13>
~~~

**Comment by rjsparks** [comment]
> I believe some BCPs have titles

Are there any where the title isn't the title of some constituent RFC? Is it always the _first_ such RFC?

In the long run, I think we should change our formal model of subseries so that each has its own actual title (even if we continue the tradition of reusing a constituent RFC title). Then 113 would keep the title it had, even if there are no longer any constituent documents. That's how I intend to implement subseries in the datatracker when we make them first-class objects. An existence proof of a BCP we already have that has a title different from an RFC that may at some point in time have been a part of it would make the motivation for that model even stronger.

**Comment by cabo** [comment]
But what is the consensus action that *makes* BCPs have a title?
The transition from zero to more than zero documents?
I don't think we have processes for actually establishing a title, so the first document gives the title?

**Comment by rjsparks** [comment]
Good question, and ask it again with STD in mind while figuring out what the better answer is
